source/to_cpp1.h

```diff
@@ -3022,13 +3130,19 @@ class cppfront
                 flush_args();
                 args.emplace(std::move(local_args));
             }
-            // Going backwards if we found Dot and there is args variable
-            // it means that it should be handled by UFCS
+            //  Going backwards if we found Dot and there is args variable
+            //  it means that it should be handled by UFCS
             else if(
                 i->op->type() == lexeme::Dot
                 && args
+                //  Disable UFCS if name lookup would hard-error (#550).
+                //  That happens when it finds that the function identifier being called is the name
+                //  of a variable with deduced type and we are in its initializer (e.g., x := y.x();)
+                //  So lower it to a member call instead, the only possible valid meaning.
+                && !lookup_finds_variable_with_placeholder_type_under_initialization(*i->id_expr)
```

Comment:
There is another case, from <https://github.com/hsutter/cppfront/issues/550#issuecomment-1834594548>.
It happens in a function expression, when lookup for the function name finds a variable outside the function expression that needs to be captured in order to be used.
UFCS should be disabled because, by definition, the function name can't be captured (#748).